msal/application.py

```diff
@@ -1642,6 +1642,11 @@ def acquire_token_for_client(self, scopes, claims_challenge=None, **kwargs):
             - an error response would contain "error" and usually "error_description".
         """
         # TBD: force_refresh behavior
+        if self.authority.tenant.lower() in ["common", "organizations"]:
+            warnings.warn(
+                "Using /common or /organizations authority "
+                "in acquire_token_for_client() is unreliable. "
+                "Please use a specific tenant instead.", DeprecationWarning)
```

Comment:
Would you happen to have an aka.ms link with more details?

Reply:
No, we do not currently have any wiki page. Not for this Client Credential Grant flow, at least.

Actually, I checked with the [MSAL .Net's wiki page on this flow](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/Client-credential-flows#code-snippet) before I created this PR. May I say that page does not really provide any extra information on this particular topic? It contains one line "do not use common...", which is kind of hidden as a comment in the code snippet. For that reason, I feel that this exception message above is sufficient.